### ISSUE TITLE
Change to Github-API Plugin breaks Jenkins builds

### ISSUE THREAD

**Issue by ziggystar** [gitbucket/gitbucket#2461]
## Change to Github-API Plugin breaks Jenkins builds

**Impacted version**: 4.32.0

A recent update of Jenkins Github-API broke our builds, and reverting the plugin update fixed them again.

```
09:53:57 09:53:57  Connecting to http://xxx/api/v3 using jenkins/******
09:53:57  org.kohsuke.github.GHFileNotFoundException: git/refs/heads/master {"message":"Not Found","documentation_url":"https://developer.github.com/v3/git/refs/#get-a-reference"}
09:53:57  	at org.kohsuke.github.GHRepository.getRef(GHRepository.java:1611)
09:53:57  	at org.jenkinsci.plugins.github_branch_source.GitHubSCMSource.retrieve(GitHubSCMSource.java:1612)
09:53:57  	at jenkins.scm.api.SCMSource.fetch(SCMSource.java:582)
09:53:57  	at org.jenkinsci.plugins.workflow.multibranch.SCMBinder.create(SCMBinder.java:98)
09:53:57  	at org.jenkinsci.plugins.workflow.job.WorkflowRun.run(WorkflowRun.java:309)
09:53:57  	at hudson.model.ResourceController.execute(ResourceController.java:97)
09:53:57  	at hudson.model.Executor.run(Executor.java:428)
09:53:57  Finished: FAILURE
```

I have tracked down the problem to recent changes in a github-api Java library. The following is just the last one in a series of changes around `getRef`.

https://github.com/hub4j/github-api/commit/9daa0df31145cb65b8dea10d3e8bc04d4a8c1128

Now I do not know who is causing the issue (the api lib, or gitbucket), because I have not deeply analyzed this.
It would be great if someone who knows his way around the GithubAPI could examine this. 
Anyway I suggest to leave this issue open until the issue of Jenkins working with Gitbucket is fixed.

**Comment by SIkebe** [comment]
Dup of #2437 ?

**Comment by skullwritter** [comment]
@SIkebe as i understood the problem, not necessarily but the #2437 might fix it.

**Comment by SIkebe** [comment]
Fixed via #2459
